### PR TITLE
Update Pydantic Model - Ignore Extra fields for `EventBase` only

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
* Followed pattern described [here](https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict.extra). 
```python
from pydantic import BaseModel, ConfigDict
class User(BaseModel):
    model_config = ConfigDict(extra='ignore')  

    name: str
``` 

* Model also defaults to `ignore` for extra fields, but there may be value in making this explicit

### CODE REVIEW

File: great_expectations_cloud/agent/models.py
Comment (Kilo59) [inline great_expectations_cloud/agent/models.py:15]: This seems dangerous to do this at the level of the `AgentBaseModel`.

Why not do it for the specific models you want this new behavior for?

Or create 2 different `AgentBaseModels`

```
AgentBaseIgnoreExtra
AgentBaseForbitExtra
```

Comment (allensallinger) [reply great_expectations_cloud/agent/models.py:15]: Łukasz had suggested that we do ignore the extra fields here: https://greatexpectationslabs.slack.com/archives/C05R8RE43CL/p1720051367833419?thread_ts=1720046306.191159&cid=C05R8RE43CL

This does make it harder for us to add more fields but it does ensure that we are not accepting in random inputs from messages.

Comment (Shinnnyshinshin) [reply great_expectations_cloud/agent/models.py:15]: Also a helpful thread here: 
https://greatexpectationslabs.slack.com/archives/C04KWBU1XDF/p1718134632349739

Will merge this PR to unblock the work, but happy to talk more next week.

**Review by Kilo59 — CHANGES_REQUESTED** [review]
I really do not think we should be in one go ignoring all fields on all models.
If we want to ignore fields for certain models, lets do that as needed.

File: great_expectations_cloud/agent/models.py
Comment (Kilo59) [reply great_expectations_cloud/agent/models.py:15]: If we want to ignore fields for certain models please create 2 separate base models and start inheriting from the `AgentBaseIgnoreExtra` where needed but otherwise inherit from `AgentBaseForbidExtra`.

Comment (Shinnnyshinshin) [reply great_expectations_cloud/agent/models.py:15]: I'm against this.  It feel like now is not the right time to put in these classes. Could risk encroaching on over-engineering

Comment (Kilo59) [reply great_expectations_cloud/agent/models.py:15]: Renaming a model is not over-engineering.

**Review by Kilo59 — APPROVED** [review]
I still think it would be a good idea to give the AgentBaseModel a more expressive name, and have 2 different versions of it. Especially when working with the runner. But I won't block on that. 
```
AgentBaseIgnoreExtra
AgentBaseForbidExtra
```

**Comment by Shinnnyshinshin** [comment]
> I still think it would be a good idea to give the AgentBaseModel a more expressive name, and have 2 different versions of it. Especially when working with the runner. But I won't block on that.
> 
> ```
> AgentBaseIgnoreExtra
> AgentBaseForbidExtra
> ```

I'd like to revisit this down the line. Happy to take responsibility if it comes back to bite us in a big way 😅

**Pull request merged**
